### ISSUE TITLE
SC1041,1042 Cannot Find 'EOF' with Output Redirection on Same Line in VSCode

### ISSUE THREAD

**Issue by adamgranthendry** [koalaman/shellcheck#2240]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1041,1042
- My shellcheck version (`shellcheck --version` or "online"): 0.7.2 (installed from "*.zip" under Releases)
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

**NOTE:** _**Works on `shellcheck.net`, but fails in VSCode with extension ShellCheck**_

OS: **Windows 10 Enterprise, Version 1909, Build 18363.1556**
VSCode: **Version 1.56.2**
VSCode Shellcheck Extension: **Version 0.14.1**

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#! /bin/sh -

my_file_='example.sh'

# Overcoming a strange Cygwin error
command cat <<EOF >"${my_file_}"
#! /bin/sh
exit 0
EOF
```

#### Here's what shellcheck currently says:
1. **shellcheck.net:** (_Run on Google Chrome, Version 90.0.4430.212 (64-bit)

```
$ shellcheck myscript
No issues detected!
```

2. **VSCode:**

```
Found 'EOF' further down, but not on a separate line.shellcheck(SC1041)

Close matches include 'EOF\r' (!= 'EOF').shellcheck(SC1042)
```

#### Here's what I wanted or expected to see:
No issues detected.

#### Please also refer to:
https://github.com/timonwong/vscode-shellcheck/issues/234

#### Notes:
I don't know if this is an issue with shellcheck or with the VSCode shellcheck extension. There is definitely a Windows element to this as it is detecting a newline simply as a carriage return. This specific error can be fixed by putting `>"${my_file_}"` in front of `<<EOF`, but I think there is a bigger problem occuring that needs addressing.

**Comment by adamgranthendry** [comment]
I can confirm this error occurs if I run `shellcheck` on the file ("script_file") from PowerShell Desktop 5.1.18362.1474, so I believe the issue with `shellcheck` itself:

```
shellcheck script_file --exclude=SC1017
```

**Comment by adamgranthendry** [comment]
This issue was solved by switching from `CRLF` to `LF` in VSCode. See [this](https://github.com/timonwong/vscode-shellcheck/issues/234#issuecomment-848301478) for solution.

**Comment by adamgranthendry** [comment]
I should also state that the syntax enforced by the `CONTRIBUTING.md` while making code easier to read also has the added (unintended) benefit of ensuring line endings are correctly set across operating systems. :)
